### ISSUE TITLE
'git lfs pull' sets all lfs files to modified and changes from lfs to git.

### ISSUE THREAD

**Issue by daxgames** [git-lfs/git-lfs#2435]
Title says it all.

Output of `git clone`, `git lfs pull` and `git lfs status`

```
bitbucket $> git clone https://****@bitbucket.org/****/test-lfs.git
Cloning into 'test-lfs'...
remote: Counting objects: 677, done.
remote: Compressing objects: 100% (341/341), done.
remote: Total 677 (delta 367), reused 544 (delta 294)
Receiving objects: 100% (677/677), 200.74 MiB | 9.88 MiB/s, done.
Resolving deltas: 100% (367/367), done.
Checking out files: 100% (222/222), done.
bitbucket $> cd test-lfs
test-lfs - master $> git lfs pull
Git LFS: (19 of 20 files) 196.82 MB / 197.08 MB
Expected OID 7633170d50ca7cfc690f1fca4dde1adc73ef0130b75fc1cb7d4517b1cfdcc5b7, got d10e697da5588ec5c1d580a8c5ee57d105fdfdaadfd9674fef73ba628cf9655c after 9977 bytes written
error: failed to fetch some objects from 'https://****@bitbucket.org/****/test-lfs.git/info/lfs'test-lfs - master! $> git lfs status                                                                                                                                                                                                                                                                [ ruby-2.3.1p112 ]
On branch master
Git LFS objects to be pushed to origin/master:


Git LFS objects to be committed:

        files/default/firmware/bxtgucver87.tar.bz2 (LFS: fc1eb19 -> Git: fc1eb19)
        files/default/firmware/kbldmcver101.tar.bz2 (LFS: 2dfb6b1 -> Git: 2dfb6b1)
        files/default/firmware/kblgucver914.tar.bz2 (LFS: 54a8350 -> Git: 54a8350)
        files/drivers/amdgpu-pro-17.10-414273.tar.xz (LFS: 74fe9b9 -> Git: 74fe9b9)
        files/manuals/e5940_P8P67.pdf (LFS: 7cc4d51 -> Git: 7cc4d51)
        files/manuals/mb_manual_ga-z170x-ud5_e.pdf (LFS: 0a39213 -> Git: 0a39213)
        files/manuals/mb_manual_ga-z270x-ud5_e.pdf (LFS: c407f62 -> Git: c407f62)
        files/mbios/mb_bios_ga-z170x-ud5_f22/Efiflash.exe (LFS: 2cc780a -> Git: 2cc780a)
        files/mbios/mb_bios_ga-z270x-ud5_f4/Efiflash.exe (LFS: 2cc780a -> Git: 2cc780a)
        files/miners/claymorev94.tar.gz (LFS: 2d5384b -> Git: 2d5384b)
        files/miners/sgminer-gm-nicehash/sgminer-5.5.5-gm-nicehash-8-linux-amd64.zip (LFS: 9fd01ce -> Git: 9fd01ce)
        files/miners/wolf-optimized-X11-X13-X15-algos.zip (LFS: 63dd4cd -> Git: 63dd4cd)
        files/scripts/sgminer-gm_startup_scripts.zip (LFS: 72bff1c -> Git: 72bff1c)
        files/tools/ADL_SDK_V10.2.zip (LFS: 6f6de21 -> Git: 6f6de21)
        files/tools/PolarisBiosEditor.exe (LFS: dbe2316 -> Git: dbe2316)
        files/tools/PolarisBiosEditor141.zip (LFS: ac52517 -> Git: ac52517)
        files/tools/SRBPolaris.zip (LFS: f4d781b -> Git: f4d781b)
        files/vbios/MSI-RX580-8GB-ARMOR-OC/MSI_RX580_8G_1-1750_Mod_bios.zip (LFS: 56a2704 -> Git: 56a2704)
        files/vbios/POWERCOLOR-TRI-X-R9-290-4GB/MBA_C6711101-100_K2_AGR_STILTMP_V2F.zip (LFS: b5490e7 -> Git: b5490e7)
        files/vbios/POWERCOLOR-TRI-X-R9-290-4GB/MBA_C6711101-100_K2_AGR_STILTMP_V48.zip (LFS: 435a8b3 -> Git: 435a8b3)

Git LFS objects not staged for commit:
```

Nothing written to `.git/lfs/objects/logs`.

**Comment by ttaylorr** [comment]
Hi @daxgames, thanks for opening this and sorry that you're having trouble. Here's what I think is going on. The error: 

> ```
> Git LFS: (19 of 20 files) 196.82 MB / 197.08 MB
> Expected OID 7633170d50ca7cfc690f1fca4dde1adc73ef0130b75fc1cb7d4517b1cfdcc5b7, got d10e697da5588ec5c1d580a8c5ee57d105fdfdaadfd9674fef73ba628cf9655c after 9977 bytes written
> error: failed to fetch some objects from 'https://****@bitbucket.org/****/test-lfs.git/info/lfs'test-lfs - master
> ```

caused the checkout to halt, so LFS didn't convert the files from pointers to objects in the working tree.

Can you double check that your LFS remote doesn't have a corrupted object, and then try again? If the object is corrupt on the remote, you can try and push over it with the object intact that you have locally.

**Comment by daxgames** [comment]
I'm new to git lfs and need a little guidance on this.  This all started when I found a git lfs file after an lfs pull that was corrupted/the wrong size.  I could not replace it with the correct file and have git lfs track it.  Every time I added and committed it was added as a normal non lfs file.  I pushed it up, all 187MB, and now I get this on a fresh clone.

Any help is appreciated.

**Comment by ttaylorr** [comment]
> Every time I added and committed it was added as a normal non lfs file.

This can happen for one of two reasons:

1. LFS's filters aren't installed. LFS relies on Git invoking a program called `git lfs filter-process` in order to convert the large files in your working tree to small files in your history. You can double check that these are installed by running:

```bash
$ git lfs install
$ git config filter.lfs.process # <- should be "git-lfs filter-process"
```

2. Your `.gitattributes` aren't invoking the Git LFS filters, which means that the files aren't being converted. You can double check this by inspecting the contents of your `.gitattributes`, and running `git lfs track` if the pattern you want to be tracked isn't.

For more information: https://git-lfs.github.com.

**Comment by daxgames** [comment]
@ttaylor Thanks I think I fixed it.  I am thinkining it was the first issue but I don't understand how.  I created this repo using git lfs so 'git lfs install' was definitely run at some point but for some reason 'git config filter.lfs.process' rutrned nothing until after I re-ran 'git lfs install'.

**Comment by eroller** [comment]
I saw similar behavior. Do we need to run "git lfs install" again when upgrading to a new version of git-lfs? How about a new version of git? Also, is there no way for git-lfs to detect if "git lfs install" has been properly run for the current user/current git/current git-lfs? It might catch a lot of errors like this that users may run into. If a command like "git lfs status" runs without error does that mean "git lfs install" does not need to be run?

**Comment by ttaylorr** [comment]
> If a command like "git lfs status" runs without error does that mean "git lfs install" does not need to be run?

I think in general this is true, (but specifically only after checkouts that contain _new_ files tracked with LFS) since LFS will have successfully converted them.

> Do we need to run "git lfs install" again when upgrading to a new version of git-lfs?

No, unless the underlying scripts change, in which case yes. I think in general you should 😄 .

> Also, is there no way for git-lfs to detect if "git lfs install" has been properly run for the current user/current git/current git-lfs?

There is not.

**Comment by PetrBodnar** [comment]
We have similiar issue during clone:
```Expected OID b675b00828707cd53dc5c130efce8a9d352e5efd8b9b29f06808810b6ebb5491, got 2c6abc2... error: failed to fetch some objects from http://192.168.1.100... ```

We check our LFS on Gitlab (we have installation of Gitlab in our local network) and got 
```  - LFS object: b675b00828707cd53dc5c130efce8a9d352e5efd8b9b29f06808810b6ebb5491: Checksum mismatch```

>Can you double check that your LFS remote doesn't have a corrupted object, and then try again? If the object is corrupt on the remote, you can try and push over it with the object intact that you have locally.

How should we push over corrupted object on Gitlab with object that we have locally?
What commands should we use?
We use Git LFS 2.4.2
Thank you!

**Comment by ttaylorr** [comment]
> How should we push over corrupted object on Gitlab with object that we have locally?
> What commands should we use?

Since Git LFS won't push an object that the server already has, you should remove the corrupted object from GitLab (or ask someone at GitLab to do so for you) and then try the `git push` as normal again.

**Comment by PetrBodnar** [comment]
> Since Git LFS won't push an object that the server already has, you should remove the corrupted object from GitLab (or ask someone at GitLab to do so for you) and then try the `git push` as normal again.

Thank you, it helps!  Solved.

**Comment by dzieciou** [comment]
I'm having similar issue.

I did fresh clone
```
git clone .../project.git
cd project/
```

There's nothing to be commited for LFS
```
$ git lfs status
On branch develop

Objects to be committed:


Objects not staged for commit:
```

Then I do 
```
$ git lfs pull
$ git lfs status
On branch develop

Objects to be committed:

        notebooks/xyz.ipynb (LFS: e9c3c9b -> Git: e9c3c9b)
        notebooks/something else.ipynb (LFS: 3a9458b -> Git: 3a9458b)
```

My .gitattributes seems correct:
```
*.ipnyb filter=lfs diff=lfs merge=lfs -text
```

And the configuration as well:
```
$ git config filter.lfs.process
git-lfs filter-process
```

What could be the reason?

The only thing that I could think of was that at some point in I moved tracked LFS files from one repository to another. Could that be a reason?

**Comment by bk2204** [comment]
Hey, @dzieciou, could you open a new issue and fill out the issue template, including the `git lfs env` output?  That would really help us track down the problem more effectively.
